src/Cake.Common/Tools/DotNetCore/Tool/DotNetCoreToolRunner.cs

```diff
@@ -53,7 +53,8 @@ public void Execute(FilePath projectPath, string command, ProcessArgumentBuilder
 
             var processSettings = new ProcessSettings();
 
-            if (projectPath != null)
+            if (processSettings.WorkingDirectory == null &&
+                projectPath != null)
             {
                 processSettings.WorkingDirectory = projectPath.GetDirectory();
             }
```

Comment:
Probably something more like this would be more in line with solving the issue.
```suggestion
            var processSettings = new ProcessSettings
            {
                WorkingDirectory = settings.WorkingDirectory ?? projectPath?.GetDirectory()
            };
```

Reply:
Aha, yes that makes more sense